src/SFA.DAS.EmployerFinance.AcceptanceTests/TestRepositories/TestTransactionRepository.cs

```diff
@@ -28,10 +31,53 @@ public class TestTransactionRepository : BaseRepository, ITestTransactionReposit
+                new PeriodEnd { PeriodEndId = "1819-R06", CalendarPeriodMonth = 1, CalendarPeriodYear = 2019, CompletionDateTime = DateTime.UtcNow, PaymentsForPeriod = "" },
+                new PeriodEnd { PeriodEndId = "1819-R07", CalendarPeriodMonth = 2, CalendarPeriodYear = 2019, CompletionDateTime = DateTime.UtcNow, PaymentsForPeriod = "" },
+                new PeriodEnd { PeriodEndId = "1819-R08", CalendarPeriodMonth = 3, CalendarPeriodYear = 2019, CompletionDateTime = DateTime.UtcNow, PaymentsForPeriod = "" },
+                new PeriodEnd { PeriodEndId = "1819-R09", CalendarPeriodMonth = 4, CalendarPeriodYear = 2019, CompletionDateTime = DateTime.UtcNow, PaymentsForPeriod = "" },
+                new PeriodEnd { PeriodEndId = "1819-R10", CalendarPeriodMonth = 5, CalendarPeriodYear = 2019, CompletionDateTime = DateTime.UtcNow, PaymentsForPeriod = "" },
+                new PeriodEnd { PeriodEndId = "1819-R11", CalendarPeriodMonth = 6, CalendarPeriodYear = 2019, CompletionDateTime = DateTime.UtcNow, PaymentsForPeriod = "" },
+                new PeriodEnd { PeriodEndId = "1819-R12", CalendarPeriodMonth = 7, CalendarPeriodYear = 2019, CompletionDateTime = DateTime.UtcNow, PaymentsForPeriod = "" },
+                new PeriodEnd { PeriodEndId = "1920-R01", CalendarPeriodMonth = 8, CalendarPeriodYear = 2019, CompletionDateTime = DateTime.UtcNow, PaymentsForPeriod = "" },
+                new PeriodEnd { PeriodEndId = "1920-R02", CalendarPeriodMonth = 9, CalendarPeriodYear = 2019, CompletionDateTime = DateTime.UtcNow, PaymentsForPeriod = "" },
+                new PeriodEnd { PeriodEndId = "1920-R03", CalendarPeriodMonth = 10, CalendarPeriodYear = 2019, CompletionDateTime = DateTime.UtcNow, PaymentsForPeriod = "" },
+                new PeriodEnd { PeriodEndId = "1920-R04", CalendarPeriodMonth = 11, CalendarPeriodYear = 2019, CompletionDateTime = DateTime.UtcNow, PaymentsForPeriod = "" });
+
+            return _employerFinanceDbContext.Value.SaveChangesAsync();
+        }
+
+        public Task CreateTransactionLines(IEnumerable<TransactionLineEntity> transactionLines)
+        {
+            _employerFinanceDbContext.Value.Transactions.AddRange(transactionLines);
+
+            return _employerFinanceDbContext.Value.SaveChangesAsync();
+        }
+
         public Task<int> GetMaxAccountId()
         {
             return _employerFinanceDbContext.Value.Database.Connection.QueryFirstAsync<int>(
-                sql: "SELECT COALESCE(MAX(AccountId), 0) FROM [employer_financial].LevyDeclaration",
+                sql: @"
+                    SELECT COALESCE(MAX(a.AccountId), 0)
+                    FROM
+                    (
+	                    SELECT AccountId FROM [employer_financial].LevyDeclaration
+                        UNION
+	                    SELECT AccountId FROM [employer_financial].Payment
+                        UNION
+	                    SELECT AccountId FROM [employer_financial].TransactionLine
```

Comment:
Couldn't we get the max AccountId just by looking at TransactionLine?

Reply:
I don't think so, it looks like tests can fail halfway through (e.g. `LevyDeclaration` data has been populated for the test account but not corresponding `TransactionLine` data). Ideally it'd come from an `Accounts` table though!